pyirf/irf/effective_area.py

```diff
@@ -85,3 +87,99 @@ def effective_area_per_energy_and_fov(
     )
 
     return effective_area(hist_selected, hist_simulated, area)
+
+
+def effective_area_3d_polar(
+    selected_events,
+    simulation_info,
+    energy_bins,
+    fov_offset_bins,
+    fov_position_angle_bins,
+):
+    """
+    Calculate effective area in bins of true energy, field of view offset, and field of view position angle.
+
+    Parameters
+    ----------
+    selected_events: astropy.table.QTable
+        DL2 events table, required columns for this function:
+        - `true_energy`
+        - `true_source_fov_offset`
+        - `true_source_fov_position_angle`
+    simulation_info: pyirf.simulations.SimulatedEventsInfo
+        The overall statistics of the simulated events
+    true_energy_bins: astropy.units.Quantity[energy]
+        The true energy bin edges in which to calculate effective area.
+    fov_offset_bins: astropy.units.Quantity[angle]
+        The field of view radial bin edges in which to calculate effective area.
+    fov_position_angle_bins: astropy.units.Quantity[radian]
+        The field of view azimuthal bin edges in which to calculate effective area.
```

Comment:
not sure if this could lead to confusion, but a 2D _azimuthal_ coordinate and _position angle_ are not quite the same: the polar azimuth starts on the positive X-axis and goes counter-clockwise, while position angle starts on the positive Y-axis and goes clockwise, so there is a 90 deg phase shift.  Perhaps just to avoid this, say "Field-of-view azimuthal (from Y-axis, toward positive x) bin edges...".      Probably this should be shown as a figure somewhere in the general PyIRF documentation...

Reply:
This is going into the same direction as the comment above about being clear how we define these coordinate systems and adding tests for the cases spelled out by GADF.

One confusing thing here is that the position angle is defined relative to the original coordinate system axes (AltAz or RaDec) but the FoV lon has the opposite direction to the original lon. Which I think makes the polar angle go counter-clockwise when looking at in a common plot of field of view coordinates where the cartesian like axes are FoV lon/lat...

Reply:
If I understand it correctly the GADF FOV position angle definition is exactly the same as the usual definition of the azimuthal angle in polar coordinates, right? (Angle increasing counter-clockwise)
So following this definition there shouldn't _technically_ be a difference between the two, although I understand that just calling it the azimuthal angle here might still cause confusion.

Since we are following the GADF conventions anyways, would it be agreeable to just change "field of view azimuthal bin edges" to "field of view position angle bin edges" and maybe showing a figure somewhere to clarify as suggested?